src/components/show-products/index.js

```diff
@@ -16,9 +16,33 @@ export const ShowProducts = () => {
   const { products, loading } = useProducts();
   const { push } = useHistory();
   const { storedValue } = useLocalStorage(LOCAL_STORAGE_LIST_TOKEN);
-  const timeFrames = TimeFrames;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [list, setList] = useState([]);
 
-  const handleOnChange = (productID) => {
+  useEffect(() => {
+    let results = products;
+
+    if (searchTerm !== '') {
+      results = products.filter(({ productName }) =>
+        productName
+          .toLowerCase()
+          .replace(/\s/g, '')
+          .normalize('NFD')
```

Comment:
For my information `NFD: Canonical Decomposition`. Today I learned this term 😮 !

Reply:
++ I also learned about that as well 🎉